schema-layer/data-structures/Data.md

```diff
@@ -0,0 +1,62 @@
+`Data` is an advanced layout for representing binary data.
+
+It is flexibible enough to support very small and very large (multi-block) binary data.
+
+```sh
+type Index [Int]
+type IndexList [Index]
+type ByteUnionList [&BytesUnion]
+
+type NestedByteListLayout struct {
+  indexes IndexList
+  parts ByteUnionList
+  algo String
+}
+
+advanced NestedByteListAdvanced {
+  rootType NestedByteListLayout
+}
+type NestedByteList bytes representation advanced NestedByteListLayoutAdvanced
+
+type ByteList [&Bytes]
+
+type ByteLinksLayout struct {
+  indexes IndexList
+  parts ByteList
+}
+
+advanced ByteLinksLayoutAdvanced {
+  rootType ByteLinksLayout
+ }
+type ByteLinks bytes representation advanced ByteLinksLayoutAdvanced
+
+type BytesUnion union {
+  | Bytes "bytes"
+  | &Bytes "bytesLink"
+  | ByteLinks "byteLinks"
+  | NestedByteList "nbl"
```

Comment:
Oh the bikeshed potential of this section!  You obviously have a "that's too long" measure in your head that made you shorten `"nestedByteList"` to `"nbl"` but unless you can be clear about how to measure that then you're going to have to defend the long-form of the other 3 from people who instinctively want to shave bytes.

I'd argue for more consistency whichever way you go. Easier to draw a firm line and defend choices.

Reply:
would still like discussion on this